### PR TITLE
Comp complaint types

### PR DESCRIPTION
**Pull request by CPage-Ten10**
Added new Ex-Gratia and Minor Misconduct workflows to the EndToEndStepDef file, and created scenarios to cover functionality that comes with those new complaint type workflows

### CODE REVIEW

File: src/test/java/com/hocs/test/glue/decs/ValidationStepDefs.java [hunk 1]
```diff
@@ -1171,96 +1171,96 @@ public void iTriggerTheErrorMessageAtTheStage(String errorMessage, String stage)
                                 safeClickOn(continueButton);
                                 break;
                             case "REASON FOR TRANSFER REQUIRED":
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/glue/decs/ValidationStepDefs.java:1173]: for all of these steps you have directly referenced WebElements from the COMP pages. Why arent you using the methods from those pages? Then we can alter just the mehtods if the page changes, and easily find all the references to those elements by seeing where the methods are used.
I know in some cases the mothods might include too many actions, and you have to directly reference a page object, but I think ideally we should keep that to a minimum.
Having 2 ways that a WebElement is referenced in the pack means we have to do more work in finding how a change effects the scenarios.

File: src/test/java/com/hocs/test/pages/comp/COMPTriage.java [hunk 2]
```diff
@@ -157,6 +165,10 @@ public void selectPermanentlyCloseCase(String yesNo) {
         clickTheButton("Confirm");
     }
 
+    public void openExGratiaAccordion() {
+        registration.openOrCloseAccordionSection("Ex-Gratia");
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/pages/comp/COMPTriage.java:169]: as COMPTriage also extends BasePage, you shouldnt need the reference to the registration object to call the method here

File: src/test/resources/features/comp/CCH.feature [hunk 3]
```diff
@@ -3,22 +3,35 @@ Feature: CCH
 
   Background:
     Given I am logged into "CS" as user "COMP_USER"
-    And I create a "COMP" case and move it to the "CCH" stage
+    When I create a "COMP" case and move it to the "CCH (From Service Triage)" stage
```
Comment (CaseyProsser) [inline src/test/resources/features/comp/CCH.feature:6]: not sure why we have switched the And and When around in this feature?
The 'When' should be the trigger for the expected scenario result, which I think was more accurate when the action step was the 'When' than the create case step

File: src/test/resources/features/comp/COMPEscalated.feature [hunk 4]
```diff
@@ -0,0 +1,82 @@
+@ServiceEscalated @COMP
+Feature: Service Escalated
+
```
Comment (CaseyProsser) [inline src/test/resources/features/comp/COMPEscalated.feature:3]: Feature name and tag need to be amended

File: src/test/resources/features/comp/COMPQA.feature [hunk 5]
```diff
@@ -0,0 +1,71 @@
+@ServiceQA @COMP
+Feature: Service QA
+
```
Comment (CaseyProsser) [inline src/test/resources/features/comp/COMPQA.feature:3]: feature name and tag need to be amended

File: src/test/resources/features/comp/COMPSend.feature [hunk 6]
```diff
@@ -0,0 +1,41 @@
+@ServiceSend @COMP
+Feature: Service Send
+
```
Comment (CaseyProsser) [inline src/test/resources/features/comp/COMPSend.feature:3]: feature name and tag need to be amended

File: src/test/java/com/hocs/test/pages/comp/COMPTriage.java (hunk 2 above)
Comment (CPage-Ten10) [reply src/test/java/com/hocs/test/pages/comp/COMPTriage.java:169]: Changed

File: src/test/resources/features/comp/CCH.feature (hunk 3 above)
Comment (CPage-Ten10) [reply src/test/resources/features/comp/CCH.feature:6]: Changed

File: src/test/resources/features/comp/COMPEscalated.feature (hunk 4 above)
Comment (CPage-Ten10) [reply src/test/resources/features/comp/COMPEscalated.feature:3]: Changed

File: src/test/resources/features/comp/COMPQA.feature (hunk 5 above)
Comment (CPage-Ten10) [reply src/test/resources/features/comp/COMPQA.feature:3]: Changed

File: src/test/resources/features/comp/COMPSend.feature (hunk 6 above)
Comment (CPage-Ten10) [reply src/test/resources/features/comp/COMPSend.feature:3]: Changed

**Pull request merged**